### PR TITLE
Geoserver Module now just collects needed config, no longer builds deployable war

### PR DESCRIPTION
**Pull request by thongsav-usgs**

### CODE REVIEW

File: coastal-hazards-geoserver/pom.xml [hunk 1]
```diff
@@ -36,56 +29,47 @@
...
-                        src/main/webapp/data/workspaces/upload,
-                        src/main/webapp/data/workspaces/draw
-                    </packagingExcludes>
-                    <failOnMissingWebXml>false</failOnMissingWebXml>
-                </configuration>
-            </plugin>
-        </plugins>
-    </build>
+	<build>
+		<finalName>geoserver</finalName>
```
Comment (jiwalker-usgs) [inline coastal-hazards-geoserver/pom.xml]: I suggest a more descriptive finalName.

File: coastal-hazards-geoserver/src/assembly/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+This zip file contains custom configuration files for Geoserver ${gs.version}.
```
Comment (jiwalker-usgs) [inline coastal-hazards-geoserver/src/assembly/README.md]: Do these tokens get replaced?

File: coastal-hazards-geoserver/src/assembly/geoserver-config-assembly.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+			<directory>${project.basedir}/src/main/webapp/data</directory>
+			<outputDirectory>data</outputDirectory>
+		</fileSet>
+		<fileSet>
+			<directory>${project.basedir}/src/assembly</directory>
+			<outputDirectory>/</outputDirectory>
+			<includes>
+				<include>README.md</include>
+			</includes>
+			<filtered>true</filtered>
```
Comment (jiwalker-usgs) [inline coastal-hazards-geoserver/src/assembly/geoserver-config-assembly.xml]: Ignore above, I see the filter happens here.

File: coastal-hazards-geoserver/src/assembly/geoserver-config-assembly.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		</fileSet>
+	</fileSets>
+	<dependencySets>
+		<dependencySet>
+			<outputDirectory>/lib</outputDirectory>
+			<useProjectArtifact>true</useProjectArtifact>
+			<unpack>false</unpack>
+			<scope>runtime</scope>
+			<includes> <!-- only include things that are not provided by tomcat/geoserver -->
+				<include>gov.usgs.cida.coastalhazards:coastal-hazards-commons</include>
```
Comment (jiwalker-usgs) [inline coastal-hazards-geoserver/src/assembly/geoserver-config-assembly.xml]: I feel like maintaining this list is going to be a nightmare.  Did you think about packaging a jar with dependencies?  I think separating the functional components from the data/configuration would be useful.  Putting them in the same zip seems to kind of break the purpose of using maven in my mind.

File: coastal-hazards-geoserver/pom.xml [hunk 5]
```diff
@@ -7,10 +7,10 @@
         <version>1.1.53-SNAPSHOT</version>
     </parent>
 
-    <artifactId>coastal-hazards-geoserver</artifactId>
-    <packaging>war</packaging>
-
-    <name>Coastal Hazards Geoserver</name>
+	<artifactId>coastal-hazards-geoserver</artifactId>
+	<packaging>pom</packaging>
```
Comment (thongsav-usgs) [inline coastal-hazards-geoserver/pom.xml]: Yeah, I wasn't sure how to make that happen. Maven said zip wasn't valid. I'll look more into it, though.

File: coastal-hazards-geoserver/pom.xml (hunk 1 above)
Comment (thongsav-usgs) [inline coastal-hazards-geoserver/pom.xml]: :+1:

File: coastal-hazards-geoserver/src/assembly/geoserver-config-assembly.xml (hunk 4 above)
Comment (thongsav-usgs) [inline coastal-hazards-geoserver/src/assembly/geoserver-config-assembly.xml]: Unfortunately it's the same issue with war/jar packaging in that the full list of dependencies is included. It was 160 ish jars versus the 10 or so listed here in the war previously, and some libs were packaged multiple times (different versions). 

It definitely will be a pain keeping it up to date, but that may be a good thing given it'll force us to revisit and review the list.

**Pull request merged**